src/image.rs

```diff
@@ -41,25 +43,27 @@ impl Image {
         file.write_all(header.as_bytes())
             .expect("Couldn't write header");
 
-        let mut to_be_written = Vec::new();
-
-        self.data
+        let to_be_written = self
+            .data
             .iter()
-            .for_each(|e| to_be_written.push((e * 255.) as u8));
-
+            .copied()
+            .map(|e| e * 255.)
+            .flat_map(|e| {
+                [e.r(), e.g(), e.b()]
+                    .iter()
+                    .map(|x| *x as u8)
+                    .collect::<Vec<u8>>()
+            })
+            .collect::<Vec<u8>>();
         file.write_all(&to_be_written).expect("Couldn't write data");
```

Comment:
why do we need to copy each element? This feels kinda strange. could we just deref it?

```suggestion
            .map(|e| (*e) * 255.)
            .flat_map(|e| {
                [e.r(), e.g(), e.b()]
                    .iter()
                    .map(|x| *x as u8)
                    .collect::<Vec<u8>>()
            })
            .collect::<Vec<u8>>();
        file.write_all(&to_be_written).expect("Couldn't write data");
```

Reply:
We can also make it a little easier to read, like this. Naming the iterated elements helps
```rust
        let buff: Vec<u8> = self
            .data
            .iter()
            .map(|pixel_array| (*pixel_array) * 255.)
            .flat_map(|pixel| [pixel.r() as u8, pixel.g() as u8, pixel.b() as u8])
            .collect();
```

Reply:
Thank you for this, I seem to have over-looked that this would involve passing through the iterator twice instead of just the once